### PR TITLE
replaced inclusion guards with pragma and ordered includes

### LINKED ISSUE #57: Apply code formatting rules
## Please describe what you would like to see

Based on the discussions in https://github.com/engelmi/orch/pull/55: 
- Exchange all `#ifndef xyz #define xyz` by `#pragma once`
- Re-order the `#include` in all source files to have the system headers first, followed by an empty line and then the local includes

### LINKED ISSUE #58: Change all headers to use "pragma once".
## Please describe what you would like to see

In systemd, ```pragma once``` serves to prevent headers from being included multiple times.  Our headers currently use inclusion guards, and we should change this to ```pragma once``` to be consistent with systemd.

### PR DESCRIPTION
**Pull request by engelmi**
This PR fixes #58 
Ths PR also fixes #57

### CODE REVIEW

File: binchihua/src/orch/orch.c [hunk 1]
```diff
@@ -1,11 +1,11 @@
-#include <stdio.h>
-#include <stdlib.h>
-
 #include "../../../libchihua-msg/include/common/common.h"
 #include "../../../libchihua-msg/include/orchestrator.h"
 #include "../ini/config.h"
 #include "opt.h"
 
+#include <stdio.h>
```
Comment (mkemel) [inline binchihua/src/orch/orch.c]: So here you have placed the system headers second, after local includes

Comment (engelmi) [reply binchihua/src/orch/orch.c]: Oh, my mistake. Shouldn't be like that. Changed.

File: binchihua/src/common/opt.h [hunk 2]
```diff
@@ -1,6 +1,7 @@
-#ifndef _BLUE_CHIHUAHUA_NODE_OPTIONS_COMMON
-#define _BLUE_CHIHUAHUA_NODE_OPTIONS_COMMON
+#pragma once
 
+// TODO: refactor
```
Comment (engelmi) [inline binchihua/src/common/opt.h]: Not sure why these macros are suddenly causing a lint error (see [here](https://github.com/engelmi/orch/actions/runs/3891920724/jobs/6643001273#step:7:10)). I disabled the check for this specific code block. If we don't want this rule, I can disable it in general. For details of the rule see [here](https://clang.llvm.org/extra/clang-tidy/checks/modernize/macro-to-enum.html)

Comment (mkemel) [reply binchihua/src/common/opt.h]: Note that it doesn't like only macros with char value. String values do not trigger this, and enum does not give us a solution for strings.
So no real way to handle this with enum.
We can leave this NOLINT comment or remove this rule, I don't think it makes much difference.

Comment (engelmi) [reply binchihua/src/common/opt.h]: Good point. And although there are workarounds to map the enum to the string, its totally unnecessary. Lets stick with disabling the linting for this block then.

Comment (mkemel) [reply binchihua/src/common/opt.h]: Yeah, but I think this is much more straightforward and readable

**Review by mkemel — APPROVED** [review]

**Pull request merged**